c++/src/Statistics.hh

```diff
@@ -41,49 +41,181 @@ namespace orc {
   };
 
 /**
+ * Internal Statistics Implementation
+ */
+
+  template <typename T>
+  class InternalStatisticsImpl {
+  private:
+    bool hasNull_;
```

Comment:
The problem with the current style is that there are no guidelines and is not consistent as well. I see some variables beginning with an `_` and some don't.
In other projects, I found the google style very helpful to read code. Especially when I have to revisit the code after some months. The style allows me to easily distinguish a class member vs a local member. Now that we plan to add the writer, the code base is only going to increase significantly. I felt this is the right time to introduce a standard moving forward. A standard helps people new to the code base too.